src/dmd/expressionsem.d

```diff
@@ -7764,6 +7765,12 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return;
         }
 
+        if (global.params.betterC)
```

Comment:
Someone using `betterC`, might still choose to provide a custom `object.d` file in which he defines e.g. `RCString` and thus _can_ use array concatenation.
Thus, wouldn't it be better to use `global.params.useTypeInfo` ?

Reply:
That would only work if array concatenation was implemented via a template in object.d. While this is a good idea, it is way beyond what this PR is.